### ISSUE TITLE
Issue while training a streaming zipformer

### ISSUE THREAD

**Issue by ezerhouni** [k2-fsa/icefall#932]
Hi everyone,

I tried training a zipformer streaming but I am seeing multiple times something like:

```Parameter Dominanting tot_sumsq module.encoder.encoders.4.out_combiner.weight1 with proportion 0.62```

after some time (before the 1st epoch), it crashes with infinite loss. Lowering ```base_lr``` seems to do the trick but I wonder if you had any idea where it might come from

**Comment by A-Raafat** [comment]
I have the same problem even after i lowered the base_lr. i also i tried to increase/decrease the warmup step but it doesnt seem to have any effect

**Comment by ezerhouni** [comment]
@A-Raafat Dividing by 5 worked for me, but I am still wondering why having such issue

**Comment by A-Raafat** [comment]
@ezerhouni i think it is something related to the optimizer function.

I tried the following and it worked out.
In the function remove_short_and_long_utt, i just filter the training data with len(tokens) to be more than  4 (which means i am expecting to have atleast 2 words in my text) and the duration to be at least more than 2 seconds .

and i no longer get the infinity loss. i dont understand why it works but i think it has something to do with computing the loss and the number of tokens

**Comment by ezerhouni** [comment]
@A-Raafat That's great ! I actually had the same intuition but didn't had time to test it out !

**Comment by ezerhouni** [comment]
@danpovey Thank you for your input. I will try to give more info soon (but I had to move myself to other tasks). I will keep you posted
